src/compiler/checker.ts

```diff
@@ -13916,14 +13916,10 @@ namespace ts {
 
                 checkDecorators(node);
                 checkSignatureDeclaration(node);
+                // TODO (weswig): Consolidate the below into checkAllCodePathsInNonVoidFunctionReturnOrThrow

```

Comment:
Looks like the remaining code is just cherry picking the case where the accessor has an implicit return and no explicit returns. Not sure why that case needs special treatmeant if it is already handled in the checkAllCodePathsEtcEtc function. What happens to the baselines if you remove it?